core/ecschema-metadata/src/test/Metadata/CustomAttributeClass.test.ts

```diff
@@ -315,6 +315,87 @@ describe("CustomAttributeClass", () => {
 
     const newDom = createEmptyXmlDocument();
 
+    it("should properly serialize custom attributes having struct inheritance", async () => {
+      const schema = {
+        $schema: ECSchemaNamespaceUris.SCHEMAURL3_2_JSON,
+        name: "TestSchema",
+        version: "1.0.0",
+        alias: "ts",
+        customAttributes: [
+          {
+            className: "TestSchema.HasColors",
+            color: {
+              r: 40,
+              g: 128,
+              b: 68,
+              hexColor: "#288044",
+            },
+          },
+        ],
+        items: {
+          HexColor: {
+            schemaItemType: "StructClass",
+            properties: [
+            {
+              type: "PrimitiveProperty",
+              typeName: "string",
+              name: "hexColor",
+            },
+            ],
+          },
+          RgbColor: {
+            schemaItemType: "StructClass",
+            baseClass: "TestSchema.HexColor",
```

Comment:
Like we discussed, the schema seems fine to me, I just think the example can be better. e.g. "An RGBColor IS a HexColor" seems confusing. Use Animal->Bird or Vehicle->Car or any common example for inheritance. Initially I think I suggested ColorRGB->ColorARGB (where ARGB just adds one alpha property)